src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String result = generateReportFromDataFile(fromFileName);
+        writeData(toFileName, result);
```

Comment:
Divide program logic: The method `getStatistic` seems to be doing two things: generating a report and writing data to a file. Consider refactoring to adhere to the single responsibility principle.

Reply:
It is condition of Task!